### ISSUE TITLE
Remove the OperationIdNounConflictingModelNames. rule

### ISSUE THREAD

**Issue by dsgouda** [Azure/azure-openapi-validator#68]
Per conversation [here](https://github.com/Azure/autorest/issues/1950#issuecomment-315830650), it looks like this rule is redundant and should be removed to avoid confusion for the users.
Documentation should be accordingly removed.

**Comment by veronicagg** [comment]
@dsgouda if the arm flag makes us append the "Operations" string every time, I agree, this rule should not be reported. Have you confirmed this is the case? I'm not clear on where this behavior is code-wise, is it AutoRest or code generator specific?

**Comment by dsgouda** [comment]
Yes I have confirmed this and we should remove this soon to avoid false positives.
